### DISCUSSION TITLE
`Parsable` vs `ParsableFactory`

### DISCUSSION THREAD

**Discussion by Acurisu** [microsoftgraph/msgraph-sdk-python#851] [Q&A] [answered]
Why does for example `Entity` implement `create_from_discriminator_value` while being a `Parsable`? Several functions such as `def get_object_value(self, factory: ParsableFactory) -> Parsable:` in the serialisers want a `ParsableFactory` but I don't see classes that inherit from `Entity` having a version of type `ParsableFactory` however they all implement `create_from_discriminator_value` (with the exception of not allowing `parse_node` being None). What is the reason for this? What is the expected way of using functions that expect `ParsableFactory`? 

Also considering `create_from_discriminator_value` being a staticmethod, shouldn't serialisers take `Type[ParsableFactory]` or similar?

**Comment by andrueastman** [comment, marked as answer]
Thanks for raising this @Acurisu 

The issue you raise is as described at https://github.com/microsoft/kiota-abstractions-python/issues/333. Please follow the issue on updates for resolving this.

> Also considering create_from_discriminator_value being a staticmethod, shouldn't serialisers take Type[ParsableFactory] or similar?

I believe you are correct here. This should be corrected once we resolve the testing of the typing end to end.
